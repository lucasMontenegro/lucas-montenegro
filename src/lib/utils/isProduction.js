import globals from "lib/utils/globals"
export default function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
    // WebPack will replace this last line with a boolean
  )
}