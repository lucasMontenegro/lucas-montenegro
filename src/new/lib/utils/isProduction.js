import global from "global"
export default function isProduction () {
  return (
    global.process ? global.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
  )
}