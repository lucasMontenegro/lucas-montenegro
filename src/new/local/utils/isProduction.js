const { default: globals } = require("new/local/utils/globals")
exports.default = function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
  )
}