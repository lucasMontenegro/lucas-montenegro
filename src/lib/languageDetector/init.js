import i18next from "i18next"
import LanguageDetector from "i18next-browser-languagedetector"
import isProduction from "lib/utils/isProduction"
export default function init (whitelist) {
  if (i18next.isInitialized) return
  i18next
    .use(LanguageDetector)
    .init({
      debug: !isProduction(),
      fallbackLng: whitelist[0],
      whitelist,
    })
}