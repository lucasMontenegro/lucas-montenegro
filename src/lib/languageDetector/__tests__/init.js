import i18next from "i18next"
import LanguageDetector from "i18next-browser-languagedetector"
import isProduction from "lib/utils/isProduction"
import init from "lib/languageDetector/init"
jest.mock(`i18next`, () => ({ __esModule: true, default: {} }))
const spy = jest.fn()
const makeSpyWrapper = name => (...args) => {
  spy(name, args)
  return i18next
}
i18next.use = makeSpyWrapper(`i18next.use`)
i18next.init = makeSpyWrapper(`i18next.init`)
jest.mock(`i18next-browser-languagedetector`, () => ({
  __esModule: true,
  default: `i18next-browser-languagedetector`,
}))
jest.mock(`lib/utils/isProduction`, () => ({ __esModule: true, default: jest.fn() }))
const languageCodes = [`foo`, `bar`]
describe(`lib/languageDetector/init`, () => {
  it(`should use the right dependency versions`, () => {
    expect(jestUtils.getDependencies([
      `i18next`,
      `i18next-browser-languagedetector`,
    ])).toMatchSnapshot()
  })
  afterEach(() => spy.mockClear())
  {
    const cases = [[true], [false]]
    const msg = `init (isInitialized true, production %j)`
    describe.each(cases)(msg, production => {
      beforeAll(() => {
        i18next.isInitialized = true
        isProduction.mockReturnValueOnce(production)
        init(languageCodes)
      })
      it(`should not call i18next methods`, () => {
        expect(spy.mock.calls).toEqual([])
      })
    })
  }
  {
    const cases = [[true], [false]]
    const msg = `init (isInitialized false, production %j)`
    describe.each(cases)(msg, production => {
      beforeAll(() => {
        i18next.isInitialized = false
        isProduction.mockReturnValueOnce(production)
        init(languageCodes)
      })
      it(`should initialize i18next`, () => {
        expect(spy.mock.calls).toMatchSnapshot()
      })
    })
  }
})