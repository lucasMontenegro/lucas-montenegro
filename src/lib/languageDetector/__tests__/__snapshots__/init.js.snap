// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/languageDetector/init init (isInitialized false, production false) should initialize i18next 1`] = `
Array [
  Array [
    "i18next.use",
    Array [
      "i18next-browser-languagedetector",
    ],
  ],
  Array [
    "i18next.init",
    Array [
      Object {
        "debug": true,
        "fallbackLng": "foo",
        "whitelist": Array [
          "foo",
          "bar",
        ],
      },
    ],
  ],
]
`;

exports[`lib/languageDetector/init init (isInitialized false, production true) should initialize i18next 1`] = `
Array [
  Array [
    "i18next.use",
    Array [
      "i18next-browser-languagedetector",
    ],
  ],
  Array [
    "i18next.init",
    Array [
      Object {
        "debug": false,
        "fallbackLng": "foo",
        "whitelist": Array [
          "foo",
          "bar",
        ],
      },
    ],
  ],
]
`;

exports[`lib/languageDetector/init should use the right dependency versions 1`] = `
Object {
  "i18next": "^17.0.18",
  "i18next-browser-languagedetector": "^3.0.3",
}
`;
